include/etcpal/cpp/timer.h

```diff
@@ -0,0 +1,314 @@
+/******************************************************************************
+ * Copyright 2019 ETC Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************
+ * This file is a part of EtcPal. For more information, go to:
+ * https://github.com/ETCLabs/EtcPal
+ ******************************************************************************/
+
+/// \file etcpal/cpp/timer.h
+/// \brief C++ wrapper and utilities for etcpal/timer.h
+
+#ifndef ETCPAL_CPP_TIMER_H_
+#define ETCPAL_CPP_TIMER_H_
+
+#include "etcpal/timer.h"
+#include "etcpal/cpp/common.h"
+
+namespace etcpal
+{
+/// \defgroup etcpal_cpp_timer timer (Timers)
+/// \ingroup etcpal_cpp
+/// \brief C++ utilities for the \ref etcpal_timer module.
+///
+/// **WARNING:** This module must be explicitly initialized before use. Initialize the module by
+/// calling etcpal_init() with the relevant feature mask:
+/// \code
+/// etcpal_init(ETCPAL_FEATURE_TIMERS);
+/// \endcode
+///
+/// **NOTE:** The constructs in this module are available in similar form in the C++ standard
+/// library using the &lt;chrono&gt; header, with the advantage that this module will work on any
+/// platform for which EtcPal is ported, including embedded RTOS platforms, using their native tick
+/// count. If your application or library does not need to run on these platforms, consider using
+/// &lt;chrono&gt; instead.
+///
+/// Provides a class for representing points in time (TimePoint) and one which implements a passive
+/// monotonic timer (Timer).
+
+/// \ingroup etcpal_cpp_timer
+/// \brief Represents a point in time.
+///
+/// More specifically, stores the number of milliseconds elapsed since an arbitrary point in the
+/// past. This number is stored as a 32-bit unsigned integer and thus wraps every 4,294,967,295
+/// milliseconds, or approximately 49.7 days. Because of this, it's recommended to only compare
```

Comment:
Are we going to have to have an int_64 version of this eventually?

Reply:
That would not be portable. Do you see a need for it?

Reply:
Not right now.  Your warnings got me thinking, that's all.